Fixed logic to match description and updated dependencies

rtkrmccaw opened the pull request:
### SUMMARY

- Updated node runtime from `12` -> `16`
- Fixed `getReleaseNotes`
  - I expect that with a prefix set, a message without that prefix does not have release notes
  - Either return an empty string or throw an error if a prefix is defined AND that prefix doesn't appear in the PR body
    - Same for if a suffix is defined

Behaviour before with `release-notes-prefix` set to `--begin--`
```
My PR description
```
The release notes for this would be "My PR description"
```
My PR description
--begin--
```
The release notes for this would be ""

With this change they now act the same. Both examples will return "" or throw an error if `require-release-notes` is `true`

### RELEASE NOTES

Fixed release notes parsing behaviour if a prefix or suffix is defined

Node runtime updated to node16 and updated other dependencies to latest possible

jefflinse commented:
Hi @rtkrmccaw, thanks for submitting this! I have a couple of other things to get to today and then I will sequence this for review and merge (along with most of the other open PRs) this afternoon.

rtkrmccaw commented:
> Hi @rtkrmccaw, thanks for submitting this! I have a couple of other things to get to today and then I will sequence this for review and merge (along with most of the other open PRs) this afternoon.

Thanks! I've fixed the problem with the action not running
Changes since you last looked:
- Updated actions/checkout from v2 -> v3 to eliminate nodejs 12 deprecation notices
- My update to `actions/github` updated the version of `octokit` and all calls need a `.rest` inserted to conform to that new version of octokit
- Noticed a deprecation warning for `@zeit/ncc` and to change to `@vercel/ncc` so that is what I did

jefflinse commented on .tool-versions:
```diff
@@ -0,0 +1 @@
+nodejs 16.18.0
```
While we're at it, would you mind adding a `.nvmrc` as well?

rtkrmccaw replied:
Will do

jefflinse reviewed (changes requested):
Changes look good, just need to merge the latest from master into your fork and then I had just one small request.

rtkrmccaw commented:
@jefflinse rebased on master and added `.nvmrc`, ready for re-review

jefflinse commented:
Appreciate the contribution, thanks Ryan! 🍻

GTB3NW commented:
@jefflinse @rtkrmccaw would it be possible to get this tagged and released please? Getting node deprecation warnings. I should be able to pin to a commit for now
